packages/cactus-plugin-keychain-aws-sm/src/test/typescript/integration/plugin-keychain-aws-sm.test.ts

```diff
@@ -0,0 +1,124 @@
+import test, { Test } from "tape-promise/tape";
+import { v4 as internalIpV4 } from "internal-ip";
+
+import {
+  Containers,
+  LocalStack,
+  K_DEFAULT_LOCALSTACK_HTTP_PORT,
+} from "@hyperledger/cactus-test-tooling";
+
+import { v4 as uuidv4 } from "uuid";
+
+import { LogLevelDesc } from "@hyperledger/cactus-common";
+
+import {
+  IPluginKeychainAwsSmOptions,
+  PluginKeychainAwsSm,
+  AwsCredentialType,
+} from "../../../main/typescript/public-api";
+
+const logLevel: LogLevelDesc = "TRACE";
+
+test("get,set,has,delete alters state as expected", async (t: Test) => {
+  const localStackContainer = new LocalStack({});
+  await localStackContainer.start();
+
+  const ci = await Containers.getById(localStackContainer.containerId);
+  const localstackIpAddr = await internalIpV4();
+  const hostPort = await Containers.getPublicPort(
+    K_DEFAULT_LOCALSTACK_HTTP_PORT,
+    ci,
+  );
+  const localstackHost = `http://${localstackIpAddr}:${hostPort}`;
+
+  test.onFinish(async () => {
+    await localStackContainer.stop();
+    await localStackContainer.destroy();
+  });
+
+  // Using awsCredentialType: AwsCredentialType.FromAwsCredentialFile
+  {
+    const options1: IPluginKeychainAwsSmOptions = {
+      instanceId: uuidv4(),
+      keychainId: uuidv4(),
+      awsEndpoint: localstackHost,
+      awsRegion: "us-east-1",
+      awsProfile: "default",
+      awsCredentialType: AwsCredentialType.LocalFile,
+      logLevel: logLevel,
+    };
+    const plugin1 = new PluginKeychainAwsSm(options1);
+
+    t.equal(plugin1.getKeychainId(), options1.keychainId, "Keychain ID set OK");
+    t.equal(plugin1.getInstanceId(), options1.instanceId, "Instance ID set OK");
+
+    const key1 = "testkey1";
+    const value1 = "testvalue1";
+
+    const hasPrior1 = await plugin1.has(key1);
+
+    t.false(hasPrior1, "hasPrior1 === false OK");
+
+    await plugin1.set(key1, value1);
+
+    const hasAfter1 = await plugin1.has(key1);
+    t.true(hasAfter1, "hasAfter1 === true OK");
+
+    const valueAfter1 = await plugin1.get(key1);
+    t.ok(valueAfter1, "valueAfter1 truthy OK");
+    t.equal(valueAfter1, value1, "valueAfter1 === value OK");
+
+    await plugin1.delete(key1);
+
+    const hasAfterDelete1 = await plugin1.has(key1);
+    t.false(hasAfterDelete1, "hasAfterDelete1 === false OK");
+
+    const valueAfterDelete1 = await plugin1.get(key1);
+    t.notok(valueAfterDelete1, "valueAfterDelete1 falsy OK");
+  }
+
+  // Using awsCredentialType: AwsCredentialType.FromCodeVariable
+  {
+    const options2: IPluginKeychainAwsSmOptions = {
+      instanceId: uuidv4(),
+      keychainId: uuidv4(),
+      awsEndpoint: localstackHost,
+      awsRegion: "us-east-1",
+      awsProfile: "default",
+      awsCredentialType: AwsCredentialType.InMemory,
+      awsAccessKeyId: "fake",
```

Comment:
@jagpreetsinghsasan Could you also look into adding a negative case for the credentials? E.g. a `plugin3` instance which has credentials that we know are invalid and then a single call to each of the keychain methods with test assertions checking that an exception is thrown with something that is indicative to the user of what is actually going wrong. Flexible on what that error message is (I assume the AWS SDK has something good for this) as long as it does let the user know that it was some sort of authorization/authentication error instead something that doesn't (think of these random errors like "cannot ready property undefined of null blah blah").

If LocalStack has a way of actually checking the AWS access key ID and secret access key to match a pre-defined value then you can use that, but if the only thing LocalStack is capable of is accepting literally anything as a valid set of credentials then you may just need to fall back on pointing the plugin to the real AWS services for `plugin3` so that we can test the negative case where all it does is fail.

Reply:
Yes, the LocalStack accepts anything as the access key ID and secret access keys.
So how should I point it to real AWS services?

Reply:
> Yes, the LocalStack accepts anything as the access key ID and secret access keys.

I get that that's the default behavior, but do they have maybe a configuration parameter in their documentation that would make it so that it actually checks the credentials against some pre-defined value?
I can imagine them maybe having a parameter that you can start the container with like `--access-key-id=myfakekey` and then it would actually only accept `myfakekey` as a valid access key id and nothing else.
I just want to make sure that you checked their documentation for something like this and if it isn't there either then we could open a feature request in the local stack repo and in the meantime fall back to using the real AWS services as you mentioned.

Reply:
I see. Let me check that out.

Reply:
The paid version (pro) of localstack supports IAM based authentication.
( I got a workaround of this in my head.. if it works, we can still do IAM access control testing)
![image](https://user-images.githubusercontent.com/56861721/118601026-61b07680-b7cf-11eb-90d4-cef79a6c983a.png)